libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -611,9 +611,9 @@ bool NavEKF2::InitialiseFilter(void)
     _framesPerPrediction = uint8_t((EKF_TARGET_DT / (_frameTimeUsec * 1.0e-6) + 0.5));
 
     // see if we will be doing logging
-    AP_Logger *dataflash = AP_Logger::get_singleton();
-    if (dataflash != nullptr) {
-        logging.enabled = dataflash->log_replay();
+    AP_Logger *logger = AP_Logger::get_singleton();
```

Comment:
No `get_singleton` calls should remain on AP_Logger!  It should all be `AP::logger()`

Reply:
needs to stay for now because we can't null check a reference very well